modules/mono/editor/GodotTools/GodotTools.Core/StringExtensions.cs

```diff
@@ -58,10 +58,19 @@ public static bool IsAbsolutePath(this string path)
                    path.StartsWith(_driveRoot, StringComparison.Ordinal);
         }
 
-        public static string ToSafeDirName(this string dirName, bool allowDirSeparator = false)
+        public static string ToSafeDirName(this string dirName, bool allowDirSeparator = false, bool assemblyName = false)
```

Comment:
Is the new parameter necessary? It seems the only other place where this method is used is in `DetermineDataDirNameForProject` but I assume the change is needed there too.

Reply:
Yes, because otherwise the other use will get a different result compared to the same call in core and it will then fail to find the data directory. (the chars are not a problem in the path, just in the AssemblyName itself.

(Or we would have to add these to the core version as well and then cause changes all over the place)